Lib/test/test_cppext.py

```diff
@@ -31,6 +36,8 @@ def test_build_cpp03(self):
     @unittest.skipIf(
         '-fsanitize' in (sysconfig.get_config_var('PY_CFLAGS') or ''),
         'test does not work with analyzing builds')
+    # the test uses pip which needs a TLS connection to PyPI
```

Comment:
I'm not sure we want to make this test reach out to PyPI to grab `setuptools`.  I think we'd be better off stashing a `setuptools` wheel (probably the one currently used by `ensurepip` :)) somewhere and installing it before using it to build the test module; that avoids network traffic and, more importantly, gives us control over the version of `setuptools` without the obligation to keep it up to date for distribution.

The other option would be to just enshrine a couple of golden compiler commands, and if they don't work :man_shrugging: oh well they probably worked on a different platform.  This test already has a ton of skip possibilities anyway.

Reply:
Adding a x-ref to https://github.com/python/cpython/pull/101039#issuecomment-1382887627 which lists the three booleans that determine which of the 6-8 (ish) possible approaches we could take here. Opinions/inputs welcome on what we want the answers to be.

From what I'm reading, you'd prefer that we don't reach out over the network. If it's preferable to just assume we can rely on setuptools and completely circumvent pip, that works. If it's preferable to have hard-coded paths compiler invocations and avoid any/all packaging tooling in the test suite, that also works for me.

To that end, I've gone ahead and added wheels for `setuptools` (and `wheel`) into the repository -- they're both necessary, assuming we want to continue having the build processes avoid reaching out to PyPI as newer versions of pip come out.

Reply:
TBH, I'd prefer to defer crafting golden compiler commands to a follow up.

It's not something that's immediately clear to me how we'd do in the current style of testing (this is, AFAICT, the only test that compiles something) and, IMO, it's likely better to have a test based off of something that's using battle-tested compiler lookup+invocation logic and then trim scope of it if we find that undesirable.